src/nupic/regions/PyRegion.cpp

```diff
@@ -82,13 +82,22 @@ extern "C"
       // clear or restore Python exceptions (see NPC-113)
       py::setRunningUnderPython();
     }
-    
+
     // Important! the following statements must be outside the PyInitialize()
     // conditional block because Python may already be initialized if used
     // through the Python bindings.
 
-    // Initialize numpy.
-    import_array();
+    // NOTE we use the function _import_array directly (just like in
+    // NumpyVector.cpp) because import_array is a macro that incorporates a
+    // return statement on failure, intended to be used directly in
+    // a python extension's init function rather than deeply nested.
+    if (_import_array() != 0)
+    {
+      // _import_array sets PyErr on failure
+      PyErr_Print();
+
+      NTA_THROW << "numpy _import_array failed";
```

Comment:
Prior to this change, execution continued as if nothing was wrong, until something else failed later, making it difficult to identify source of failure quickly.